server/athenian/api/controllers/miners/github/precomputed_prs/done_prs.py

```diff
@@ -0,0 +1,760 @@
+        await load_branch_commit_dates(branches, meta_ids, mdb)
+        return branches
+
+    ghdprf = GitHubDonePullRequestFacts
+    tasks = [
+        pdb.fetch_all(select([ghdprf.pr_node_id])
+                      .where(and_(ghdprf.repository_full_name.in_(repos),
+                                  ghdprf.acc_id == account,
+                                  ghdprf.release_match.like("%|%")))),
+        fetch_branches(),
+        fetch_precomputed_commit_history_dags(repos, account, pdb, cache),
+    ]
+    rows, branches, dags = await gather(*tasks, op="fetch prs + branches + dags")
+    pr_node_ids = [r[0] for r in rows]
+    del rows
+    tasks = [
+        mdb.fetch_all(select([PullRequest.merge_commit_sha, PullRequest.node_id])
+                      .where(and_(PullRequest.node_id.in_(pr_node_ids),
+                                  PullRequest.acc_id.in_(meta_ids)))),
+        fetch_repository_commits(
+            dags, branches, BRANCH_FETCH_COMMITS_COLUMNS, True, account, meta_ids,
+            mdb, pdb, cache),
+    ]
+    del pr_node_ids
+    pr_merges, dags = await gather(*tasks, op="fetch merges + prune dags")
+    accessible_hashes = np.sort(np.concatenate([dag[0] for dag in dags.values()]))
+    merge_hashes = np.sort(np.fromiter((r[0] for r in pr_merges), "S40", len(pr_merges)))
+    found = searchsorted_inrange(accessible_hashes, merge_hashes)
+    dead_indexes = np.nonzero(accessible_hashes[found] != merge_hashes)[0]
+    dead_pr_node_ids = [None] * len(dead_indexes)
+    for i, dead_index in enumerate(dead_indexes):
+        dead_pr_node_ids[i] = pr_merges[dead_index][1]
+    del pr_merges
+    with sentry_sdk.start_span(op="delete force push dropped prs",
+                               description=str(len(dead_indexes))):
+        await pdb.execute(
+            delete(ghdprf)
+            .where(and_(ghdprf.pr_node_id.in_(dead_pr_node_ids),
+                        ghdprf.release_match != ReleaseMatch.force_push_drop.name)))
+    return dead_pr_node_ids
+
+
+def _flatten_set(s: set) -> Optional[Any]:
+    if not s:
+        return None
+    assert len(s) == 1
+    return next(iter(s))
+
+
+# TODO: these have to be removed, these are here just for keeping backward-compatibility
+# without the need to re-write already all the places these functions are called
+load_precomputed_done_candidates = DonePRFactsLoader.load_precomputed_done_candidates
+load_precomputed_done_facts_all = DonePRFactsLoader.load_precomputed_done_facts_all
+load_precomputed_done_facts_filters = DonePRFactsLoader.load_precomputed_done_facts_filters
+load_precomputed_done_facts_ids = DonePRFactsLoader.load_precomputed_done_facts_ids
+load_precomputed_done_facts_reponums = DonePRFactsLoader.load_precomputed_done_facts_reponums
+load_precomputed_pr_releases = DonePRFactsLoader.load_precomputed_pr_releases
+load_precomputed_done_timestamp_filters = DonePRFactsLoader.load_precomputed_done_timestamp_filters
```

Comment:
These will be removed in a following PR similarly as done for `merged_prs` in #1436.